src/util/tab-events.js

```diff
@@ -1,56 +1,55 @@
 import eventToPromise from './event-to-promise'
 
+const tabChangedEvents = [
+    {
+        event: browser.webNavigation.onCommitted,
+        filter: (details)=>(details.tabId == tabId && details.frameId == 0),
+        reason: "Tab URL changed before loading was complete"
+    },
+    {
+        event: browser.webNavigation.onHistoryStateUpdated,
+        filter: (details)=>(details.tabId == tabId && details.frameId == 0),
+        reason: "Tab URL changed before loading was complete"
+    },
+    {
+        event: browser.tabs.onRemoved,
+        filter: (closedTabId) => (closedTabId === tabId),
+        reason: {message: "Tab was closed before loading was complete."},
+    }
+];
+
+
 // Resolve if or when the page DOM is loaded (document.readyState==='interactive')
 // Rejects if it is closed before that.
 // XXX Needs host permission on the tab
-// TODO reject if loading of the current page is aborted. (#15)
 export function whenPageDOMLoaded({tabId}) {
-    // This more obvious approach can get stuck in limbo, as there is no
-    // tab.status==='interactive'; it is either 'loading' or 'complete'.
-    // return browser.tabs.get(tabId).then(tab => {
-    //     if (tab.status === 'complete') // XXX Tests the wrong thing.
-    //         return // Resolve directly
-    //
-    //     return eventToPromise({
-    //         resolve: {
-    //             event: browser.webNavigation.onDOMContentLoaded,
-    //             filter: ({tabId: eventTabId}) => (eventTabId===tabId),
-    //         },
-    //         reject: {
-    //             event: browser.tabs.onRemoved,
-    //             filter: (closedTabId) => (closedTabId === tabId),
-    //             reason: {message: "Tab was closed before DOM was loaded."},
-    //         },
-    //     })
-    // })
```

Comment:
Removing the bulk of this commented code is fine, but I would still add an explanation why we use this `executeScript` approach: there is no tab.status==='interactive'.